plugins/woocommerce/includes/wc-product-functions.php

```diff
@@ -1834,3 +1835,30 @@ function wc_product_attach_featured_image( $attachment_id, $product = null, $sav
 	}
 }
 add_action( 'add_attachment', 'wc_product_attach_featured_image' );
+
+/**
+ * Delete related product transients when a product is saved.
+ * This is necessary because changing one product might affect many related products.
+ *
+ * @since 9.8.1
+ * @param int $product_id Product ID.
+ */
+function wc_delete_related_product_transients( $product_id ) {
+	global $wpdb;
+
+	// Delete all related product timeout transients
+	$wpdb->query( 
+		$wpdb->prepare( 
+			"DELETE FROM {$wpdb->options} WHERE option_name LIKE %s",
+			$wpdb->esc_like( '_transient_timeout_wc_related_' ) . '%'
+		)
+	);
+
+	// Also delete the actual transient entries
+	$wpdb->query( 
+		$wpdb->prepare( 
+			"DELETE FROM {$wpdb->options} WHERE option_name LIKE %s",
+			$wpdb->esc_like( '_transient_wc_related_' ) . '%'
+		)
+	);
```

Comment:
Not a big deal, but wouldn't it be better to do this in one query instead of two? Or is there a reason to prefer doing two queries?

```suggestion
	// Delete both timeout and actual transient entries in one query
	$wpdb->query( 
	    $wpdb->prepare( 
	        "DELETE FROM {$wpdb->options} WHERE option_name LIKE %s OR option_name LIKE %s",
	        $wpdb->esc_like( '_transient_timeout_wc_related_' ) . '%',
	        $wpdb->esc_like( '_transient_wc_related_' ) . '%'
	    )
	);
```

Reply:
Good idea, will do!